collector/cpu_linux.go

```diff
@@ -83,8 +114,64 @@ func NewStatCollector() (Collector, error) {
 	}, nil
 }
 
+func (c *cpuCollector) Update(ch chan<- prometheus.Metric) error {
+	err1 := c.UpdateStat(ch)
+	err2 := c.UpdateCpufreq(ch)
+	if err1 != nil {
+		return err1
+	}
+	if err2 != nil {
+		return err2
+	}
```

Comment:
I always want to report an error if either breaks, but I always want to try both.... Doesn't reusing the variable shadow the first error?